.github/workflows/ci.yaml

```diff
@@ -13,5 +13,6 @@ jobs:
     - uses: actions/setup-go@v2.1.3
       with:
         go-version: 1.16.x
+    - run: go get github.com/onsi/ginkgo/ginkgo@v1.16.4
```

Comment:
let's follow the same pattern that's used for tools like controller-gen and kustomize to install a known version rather than assuming that a dev has an appropriate version installed.

Reply:
@scothis Thanks for the suggestion. I have changed it accordingly.
I squashed the commits as the older commits were not useful. Please take review again.